backend/src/StdLibExecution/Libs/String.fs

```diff
@@ -972,6 +972,28 @@ let fns : List<BuiltInFn> =
         | _ -> incorrectArgs ())
       sqlSpec = NotYetImplemented
       previewable = Pure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "String" "indexOf" 0
+      typeParams = []
+      parameters =
+        [ Param.make "str" TString "The string to search in"
+          Param.make
+            "searchFor"
+            TString
+            "The string to search for within <param str>" ]
+      returnType = TInt
+      description =
+        "Returns the index of the first occurrence of <param searchFor> in <param str>, or -1 if <param searchFor> does not occur."
```

Comment:
(can be done in another PR) returning an Option<Int> might make more sense for this fn